docs/madr/decisions/032-add-inbound-in-runtime.md

```diff
@@ -0,0 +1,142 @@
+* Updating changed data plane proxy with new certificate
+* Updating all clients with new endpoint targeting changed data plane proxy
+
+If second applies before the first, it can result in broken traffic.
+
+This can happen with ArgoCD and Blue Green rollout.
+First the pod is selected by `service-preview` and if it passes the analysis the pod is targeted by the `service-active`.
+
+Here is an example of `redis` from `kuma-demo` going through blue-green promotion 
+```yaml
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis
+  namespace: kuma-demo
+spec:
+  ports:
+    - port: 6379
+      protocol: TCP
+      targetPort: 6379
+  selector:
+    app: redis
+    rollouts-pod-template-hash: 778dbdddff
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-preview
+  namespace: kuma-demo
+spec:
+  ports:
+    - port: 6379
+      protocol: TCP
+      targetPort: 6379
+  selector:
+    app: redis
+    rollouts-pod-template-hash: 646998df5c
+```
+Once it's finished, `redis` will change selector to `rollouts-pod-template-hash: 646998df5c` and `redis-preview` will keep `rollouts-pod-template-hash: 646998df5c` selector. 
+
+## Decision Drivers
+
+* Enable users to use ArgoCD blue-green rollout without traffic interruption
+
+## Considered Options
+
+* Wait for certs using Dataplane Insights
+* Wait for certs using XDS callbacks
+* Ignore labels in selector to add ignored listener
+
+## Decision Outcome
+
+Chosen option: "Ignore labels in selector to add ignored listener". While it's the hardest to explain. It has no impact on the perf.
+
+## Pros and Cons of the Options
+
+To consider options, we need to first introduce new concept of "Ignored" listener.
+Instead of `healthy: false/true` we also need another state of the listener.
+
+```yaml
+type: Dataplane
+mesh: default
+name: redis-646998df5c-nzx4b.kuma-demo
+networking:
+  address: 10.42.0.25
+  inbound:
+    - port: 6379
+      tags:
+        app: redis
+        kuma.io/service: redis_kuma-demo_svc_6379
+        rollouts-pod-template-hash: 646998df5c
+      state: Ready | NotReady | Ignored
+```
+
+* Ready - listener is ready to receive the traffic (equivalent of existing `healthy: true`)
+* NotReady - listener is not ready to receive the traffic (equivalent of existing `healthy: false`)
+* Ignored - listener is not created. It cannot be targeted by policies. It will however receive a cert with `kuma.io/service` of this listener.
+
+**Why we need Ignored state. Can't we just use NotReady?**
+* Targeting outbound. Let say we have active `backend` and `backend-preview`. We want to redirect the traffic from `backend-preview` to `redis-staging`.
+  We create traffic policies `backend->redis` and `backend-preview->redis-staging`. If `backend-preview` gets NotReady listener for `backend` and ready for `backend-preview`.
+  It may match either `backend->redis` or `backend-preview->redis-staging` and cause unexpected behavior.
```

Comment:
These sentences are unclear to me. I think they should be one sentence. Could you make it a little bit clearer and easier to understand?

Reply:
Please take a look if it's more clear. If not, it would be great if you would say what is not clear about it, thanks!

Reply:
Yup, now I got it.